enos/modules/install_packages/scripts/install-packages.sh

```diff
@@ -9,41 +9,114 @@ fail() {
   exit 1
 }
 
-[[ -z "$RETRY_INTERVAL" ]] && fail "RETRY_INTERVAL env variable has not been set"
-[[ -z "$TIMEOUT_SECONDS" ]] && fail "TIMEOUT_SECONDS env variable has not been set"
-[[ -z "$PACKAGES" ]] && fail "PACKAGES env variable has not been set"
+[[ -z "${RETRY_INTERVAL}" ]] && fail "RETRY_INTERVAL env variable has not been set"
+[[ -z "${TIMEOUT_SECONDS}" ]] && fail "TIMEOUT_SECONDS env variable has not been set"
+[[ -z "${PACKAGES}" ]] && fail "PACKAGES env variable has not been set"
+[[ -z "${PACKAGE_MANAGER}" ]] && fail "PACKAGE_MANAGER env variable has not been set"
 
 install_packages() {
-  if [ "$PACKAGES" = "__skip" ]; then
+  if [[ "${PACKAGES}" = "__skip" ]]; then
     return 0
-  fi
+  fi 
 
-  echo "Installing Dependencies: $PACKAGES"
-  if [ -f /etc/debian_version ]; then
-    # Do our best to make sure that we don't race with cloud-init. Wait a reasonable time until we
-    # see ec2 in the sources list. Very rarely cloud-init will take longer than we wait. In that case
-    # we'll just install our packages.
-    grep ec2 /etc/apt/sources.list || true
-
-    cd /tmp
-    sudo apt update
-    # shellcheck disable=2068
-    sudo apt install -y ${PACKAGES[@]}
-  else
-    cd /tmp
-    # shellcheck disable=2068
-    sudo yum -y install ${PACKAGES[@]}
-  fi
+  set -x
+  echo "Installing Dependencies: ${PACKAGES}"
+
+  # Use the default package manager of the current Linux distro to install packages
+  case $PACKAGE_MANAGER in
+
+    "apt")
+      sudo apt update
+      for package in ${PACKAGES}; do
+        if dpkg -s "${package}"; then
+          continue
+        else
+          echo "Installing ${package}"
+          sudo apt install -y "${package}"
+        fi
+      done
+      ;;
+
+    "yum")
+      for package in ${PACKAGES}; do
+        # if sudo yum list installed | rpm -q "${package}"; then
+        if rpm -q "${package}"; then
+          continue
+        else
+          echo "Installing ${package}"
+          sudo yum -y install "${package}"
+        fi
+      done
+      ;;
+
+    "zypper")
+      cd /tmp
+      sudo zypper --gpg-auto-import-keys ref
+      for package in ${PACKAGES}; do
+        if rpm -q "${package}"; then
+          continue
+        else
+          echo "Installing ${package}"
+          sudo zypper --non-interactive install "${package}"
+          date
+        fi
+        sudo zypper search -i
+      done
+      ;;
+
+    *)
+      fail "No matching package manager provided."
+      ;;
+
+  esac
+
+#   # Use the default package manager of the current Linux distro to install packages
```

Comment:
Is this comment left here intentionally?

Reply:
Nope 🙈 removed.